src/js/utils.js

```diff
@@ -16,6 +16,27 @@ exports.is_sample_genetically_altered = function is_sample_genetically_altered(d
   || datum.protein !== undefined;
 };
 
+exports.makeD3SVGElement = function(tag) {
+  return d3.select(document.createElementNS('http://www.w3.org/2000/svg', tag));
+};
+
+exports.stableSort = function(arr, cmp) {
```

Comment:
I guess general is the wrong word, since a well-defined comparator will correspond one-to-one with a metric function given by the final sort order.

But for some reason I feel like it can be easier to specify a comparator than to explicitly specify the metric. Maybe not, what do you think?